### PR TITLE
Add Documentation for Configuration

### LINKED ISSUE #76: Fix Introduction in the Documentation
In the 'Running the Framework' section the ```-DANALYSIS=config -DSAMPLES=samples``` flags need to be specified / how the config and samples are handled is unclear.

### PR DESCRIPTION
**Pull request by harrypuuter**
Resolves #76

### CODE REVIEW

**Comment by harrypuuter** [comment]
This PR updates the Docs including an example of all functions provided by the configuration class. Also resolves #76

File: code_generation/systematics.py [hunk 1]
```diff
@@ -30,35 +30,38 @@
...
-    set of configuration parameters. A dummy shift looks like this:
-    SystematicShift(
-            name="shiftname",
-            shift_config={
-                ("scope1", "scope2"): {
-                    "paramter1": True,
-                    "paramter2": 42.0,
+    """Class containing systematic shifts. A systematic shift is a variation of a set of configuration parameters.
+
+    A dummy shift looks like this::
```
Comment (nshadskiy) [inline code_generation/systematics.py:35]: typo "::"

File: CMakeLists.txt [hunk 2]
```diff
@@ -149,7 +149,17 @@ function(find_python_package PYPINAME NAME MIN_VERSION)
...
+execute_process(COMMAND correction config --cmake
+    OUTPUT_VARIABLE CORRECTION_LIB_ARGS
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+message(STATUS "Correctionline library path: ${CORRECTION_LIB_ARGS}")
+string(REPLACE -Dcorrectionlib_DIR= "" CORRECTIONLIBPATH ${CORRECTION_LIB_ARGS})
+message(STATUS "Correctionline library path: ${CORRECTIONLIBPATH}")
+find_package(correctionlib REQUIRED PATHS ${CORRECTIONLIBPATH})
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads)
+find_package(ZLIB)
```
Comment (nshadskiy) [inline CMakeLists.txt:162]: The `correctionlib` installation should be excluded here for now, since it is not ready yet.

File: CMakeLists.txt [hunk 3]
```diff
@@ -199,7 +209,7 @@ foreach(FILENAME ${FILELIST})
     add_executable(${TARGET_NAME} ${FULL_PATH})
     # target_precompile_headers(${TARGET_NAME} PRIVATE jets.hxx lorentzvectors.hxx metfilter.hxx pairselection.hxx quantities.hxx)
     target_include_directories(${TARGET_NAME} PRIVATE ${CMAKE_SOURCE_DIR} ${ROOT_INCLUDE_DIRS})
-    target_link_libraries(${TARGET_NAME} ROOT::ROOTVecOps ROOT::ROOTDataFrame ROOT::RooFit logging)
+    target_link_libraries(${TARGET_NAME} ROOT::ROOTVecOps ROOT::ROOTDataFrame ROOT::RooFit logging correctionlib)
```
Comment (nshadskiy) [inline CMakeLists.txt:212]: Exclude `correctionlib` for now,

**Review by nshadskiy — APPROVED** [review]
Looks good!

File: code_generation/systematics.py (hunk 1 above)
Comment (harrypuuter) [reply code_generation/systematics.py:35]: This is doctring formatting, so the block below is rendered as a codeblock in the documentation

File: docs/sphinx_source/py_configuration.rst [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        ["mm", "mt'],
+        {
+            "applyRecoilCorrections":
+                SampleModifier({"wj": True}, default=False),
+        },
+    )
+
+In this example, the parameter ``applyRecoilCorrections`` is added to the parameter list of the producers ``mm`` and ``mt``. Due to the configuration of the modifier, the parameter will be set to ``True`` for the samples of the type ``wj``. For all other samples, the default value ``False`` is set.
+
+Internally, the revolving of modifiers is done using the :py:func:`~code_generation.configuration.Configuration.resolve_modifiers` function, which is called automatically by the :py:func:`~code_generation.configuration.Configuration.add_config_parameters` function.
```
[the remark this answers is not in the document]
Comment (harrypuuter) [reply docs/sphinx_source/py_configuration.rst:122]: changed

File: docs/sphinx_source/py_configuration.rst [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        "mm",
+        [
+            muons.GoodMuons,
+            muons.VetoMuons,
+            pairselection.MMPairSelection,
+            pairselection.GoodMMPairFilter,
+        ]
+    )
+
+In this example, the producers ``GoodMuons``, ``VetoMuons``, ``MMPairSelection`` and ``GoodMMPairFilter`` are added to the list of producers ``mm``. The producers added here can be both a :py:class:`~code_generation.producer.Producer` or a :py:class:`~code_generation.producers.ProducerGroup`. Also, the oder of the producers is not important. Using the :py:func:`~code_generation.configuration.Configuration.optimize` function, the producers ordering will be optimized, such that filters are always run first, and that producers that depend on other producers are run after the producers that they depend on.
```
[the remark this answers is not in the document]
Comment (harrypuuter) [reply docs/sphinx_source/py_configuration.rst:166]: changed

File: CMakeLists.txt (hunk 2 above)
Comment (harrypuuter) [reply CMakeLists.txt]: changed

**Pull request merged**